crates/autopilot/src/infra/persistence/mod.rs

```diff
@@ -382,6 +387,129 @@ impl Persistence {
         Ok(solution)
     }
 
+    /// Computes solvable orders based on the latest observed block number,
+    /// order creation timestamp, and minimum validity period.
+    pub async fn solvable_order_after(
+        &self,
+        current_orders: HashMap<domain::OrderUid, model::order::Order>,
+        after_timestamp: DateTime<Utc>,
+        after_block: u64,
+        min_valid_to: u32,
+    ) -> anyhow::Result<boundary::SolvableOrders> {
+        let after_block = i64::try_from(after_block).context("block number value exceeds i64")?;
+        let mut tx = self.postgres.pool.begin().await.context("begin")?;
+        // Set the transaction isolation level to REPEATABLE READ
+        // so all the SELECT queries below are executed in the same database snapshot
+        // taken at the moment before the first query is executed.
+        sqlx::query("SET TRANSACTION ISOLATION LEVEL REPEATABLE READ")
+            .execute(tx.deref_mut())
+            .await?;
+
+        // Find order uids for orders that were updated after the given block.
+        let updated_order_uids: Vec<database::OrderUid> = {
+            let _timer = Metrics::get()
+                .database_queries
+                .with_label_values(&["updated_order_uids"])
+                .start_timer();
+
+            database::orders::updated_order_uids_after(&mut tx, after_block)
```

Comment:
Coming to think about it, cannot we do this query and the next ones in a single roundtrip? or at least to minimize it? we have 3/4 roundtrips to the database.

Reply:
Please elaborate. Not sure which queries could be combined.

Reply:
sure! so, my concern is that we do these two queries to the orders database:

- `updated_order_uids_after`
- `open_orders_by_time_and_uids`

The thing is that the output of `updated_order_uids_after` is used as an input for `open_orders_by_time_and_uids`. So, we potentially could combine those queries into one roundtrip (of course, the query would be relatively complex), but we do have already such complexity levels in queries (e.g. see `ORDERS_SELECT`).

Reply:
Nice idea 👍  Unified queries.